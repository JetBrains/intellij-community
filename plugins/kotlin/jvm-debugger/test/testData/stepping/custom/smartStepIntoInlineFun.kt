package smartStepIntoInlinedFunLiteral

fun main(args: Array<String>) {
    val array = arrayOf(1, 2)
    val myClass = MyClass()

    //Breakpoint!
    test()
    // STEP_OVER: 1
    // SMART_STEP_INTO_BY_INDEX: 1
    myClass.f1 { test() }
           .f2 { test() }

    // RESUME: 1
    //Breakpoint!
    test()
    // STEP_OVER: 1
    // SMART_STEP_INTO_BY_INDEX: 3
    myClass.f1 { test() }
        .f2 { test() }

    // RESUME: 1
    //Breakpoint!
    test()
    // STEP_OVER: 1
    // SMART_STEP_INTO_BY_INDEX: 1
    array.map {
        it *2
    }

    // RESUME: 1
    //Breakpoint!
    test()
    // STEP_OVER: 1
    // SMART_STEP_INTO_BY_INDEX: 1
    array.map { it * 2 }
         .filter {
             it > 2
         }

    // RESUME: 1
    //Breakpoint!
    test()
    // STEP_OVER: 1
    // SMART_STEP_INTO_BY_INDEX: 3
    array.map { it * 2 }
        .filter {
            it > 2
        }

    // RESUME: 1
    //Breakpoint!
    test()
    // STEP_OVER: 1
    // SMART_STEP_INTO_BY_INDEX: 1
    myClass.f1 { test() }.f2 { test() }

    // RESUME: 1
    //Breakpoint!
    test()
    // STEP_OVER: 1
    // SMART_STEP_INTO_BY_INDEX: 3
    myClass.f1 { test() }.f2 { test() }
}

class MyClass {
    inline fun f1(f1Param: () -> Unit): MyClass {
        test()
        f1Param()
        return this
    }

    inline fun f2(f1Param: () -> Unit): MyClass {
        test()
        f1Param()
        return this
    }
}

fun test() {}

// IGNORE_K2