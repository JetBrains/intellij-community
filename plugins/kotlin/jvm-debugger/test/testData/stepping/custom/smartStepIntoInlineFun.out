LineBreakpoint created at smartStepIntoInlineFun.kt:8
LineBreakpoint created at smartStepIntoInlineFun.kt:16
LineBreakpoint created at smartStepIntoInlineFun.kt:24
LineBreakpoint created at smartStepIntoInlineFun.kt:33
LineBreakpoint created at smartStepIntoInlineFun.kt:43
LineBreakpoint created at smartStepIntoInlineFun.kt:53
LineBreakpoint created at smartStepIntoInlineFun.kt:60
Run Java
Connected to the target VM
smartStepIntoInlineFun.kt:8
smartStepIntoInlineFun.kt:11
smartStepIntoInlineFun.kt:68
smartStepIntoInlineFun.kt:16
smartStepIntoInlineFun.kt:19
smartStepIntoInlineFun.kt:74
smartStepIntoInlineFun.kt:24
smartStepIntoInlineFun.kt:27
_Arrays.!EXT!
smartStepIntoInlineFun.kt:33
smartStepIntoInlineFun.kt:36
_Arrays.!EXT!
smartStepIntoInlineFun.kt:43
smartStepIntoInlineFun.kt:46
_Collections.!EXT!
smartStepIntoInlineFun.kt:53
smartStepIntoInlineFun.kt:56
smartStepIntoInlineFun.kt:68
smartStepIntoInlineFun.kt:60
smartStepIntoInlineFun.kt:63
smartStepIntoInlineFun.kt:74
Disconnected from the target VM

Process finished with exit code 0
